### PR TITLE
More descriptive error messages for elasticache provider

### PR DESCRIPTION
**Pull request by bsquizz**
The current error message of `Missing dependencies: [- in-memory-db-secret: sources-api]` is a little unclear.

Technically the problem is that we're missing the secret named `in-memory-db` in the sources-api app's namespace -- modified the error messages to make this more obvious.

### CODE REVIEW

File: config/crd/bases/cyndi.cloud.redhat.com_cyndipipelines.yaml [hunk 1]
```diff
@@ -1,38 +1,13 @@
 
 ---
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.3.0
+    controller-gen.kubebuilder.io/version: v0.5.0
```
Comment (BlakeHolifield) [inline config/crd/bases/cyndi.cloud.redhat.com_cyndipipelines.yaml:7]: Do you need this for cyndi to generate correctly? Trying to determine if you have a different version or if this was an intended version bump.

File: config/crd/bases/cyndi.cloud.redhat.com_cyndipipelines.yaml [hunk 2]
```diff
@@ -1,38 +1,13 @@
 
 ---
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
```
Comment (BlakeHolifield) [inline config/crd/bases/cyndi.cloud.redhat.com_cyndipipelines.yaml:3]: Same sort of question here. Seems like you get this from the controller-gen version.

File: config/crd/bases/cloud.redhat.com_clowdapps.yaml [hunk 3]
```diff
@@ -1,26 +1,13 @@
 
 ---
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.3.0
+    controller-gen.kubebuilder.io/version: v0.5.0
```
Comment (BlakeHolifield) [inline config/crd/bases/cloud.redhat.com_clowdapps.yaml:7]: Here as well

File: config/crd/bases/cloud.redhat.com_clowdenvironments.yaml [hunk 4]
```diff
@@ -1,26 +1,13 @@
 
 ---
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.3.0
+    controller-gen.kubebuilder.io/version: v0.5.0
```
Comment (BlakeHolifield) [inline config/crd/bases/cloud.redhat.com_clowdenvironments.yaml:7]: And here

**Review by BlakeHolifield — COMMENTED** [review]
Good change, just questions about the config gen updates.

**Comment by psav** [comment]
Until we have updated everything in the repo, can you please use the same operator sdk version so that we don't get 7000 lines of file changes?

**Review by psav — CHANGES_REQUESTED** [review]
Need to keep the changes to only relevant ones

**Comment by bsquizz** [comment]
Ugh sorry, too used to using `git add -u`

**Review by BlakeHolifield — APPROVED** [review]

File: controllers/cloud.redhat.com/providers/inmemorydb/elasticache.go [hunk 5]
```diff
@@ -50,7 +55,7 @@ func (e *elasticache) Provide(app *crd.ClowdApp, config *config.AppConfig) error
 	if !found {
 		return &errors.MissingDependencies{
 			MissingDeps: map[string][]string{
-				"in-memory-db-secret": {app.Name},
+				"secret": {fmt.Sprintf("name: %s, namespace: %s", secretName, app.Namespace)},
```
Comment (psav) [inline controllers/cloud.redhat.com/providers/inmemorydb/elasticache.go:58]: Calling this just `secret` is too generic - we could have multiple "secrets" that are missing dependencies, we either need a more intelligent missing dependency map where we can "add" items to a category, or keep it with a more unique name.

**Review by psav — CHANGES_REQUESTED** [review]

File: controllers/cloud.redhat.com/providers/inmemorydb/elasticache.go (hunk 5 above)
Comment (bsquizz) [reply controllers/cloud.redhat.com/providers/inmemorydb/elasticache.go]: Ok.. I like your first idea, but sounds like something we probably need a bit more design around ... just reverted the name for now.

**Review by psav — APPROVED** [review]

**Pull request merged**